docs/08-docker/02-windows-docker-images.mdx

```diff
@@ -17,70 +17,94 @@ found in
+IL2CPP C++ code builder is unable to build C++ code. In order to build C++ code for Windows Desktop, you must have one of these installed:
+ * Visual Studio 2022 or newer with C++ compilers and Windows 10 SDK (recommended)
+ * Visual Studio 2019 with C++ compilers and Windows 10 SDK
+ * Visual Studio 2017 with C++ compilers and Windows 10 SDK
+ * Visual Studio 2015 with C++ compilers and Windows 10 SDK
+Visual Studio 2017 (or newer) is detected using `vswhere.exe` as well as VSCOMNTOOLS environment variables.
+Visual Studio 2015 is detected by looking at "SOFTWARE\Microsoft\VisualStudio\14.0_Config\InstallDir" in the registry as well as VSCOMNTOOLS environment variables.
+Windows 10 SDK is detected by looking at "SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0\InstallationFolder" in the registry.
+Unable to detect any compatible Visual Studio installation!
+ * Found Visual Studio 2022 installation without C++ tool components
+Windows 10 SDK is not installed. You can install from here: https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk/
+```
 
-### Injecting VC++ into the image
+There are two options to avoid this obstacle:
 
-**To pass VC++ from a host system, two steps are necessary**
+1. (*Recommended*) Inject/mount Microsoft Visual Studio with VC++ and Windows 10 SDK from the host system into the
+   container during runtime.
+2. Build a custom image and push it to a docker registry.
 
-_Add to the system registry (in a container) path and version to Windows SDK_
+#### Option 1. Injecting VC++ from the host system into the container
 
-```powershell
-$windowsSdkDirectory = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0"
+> These instructions have been validated with Visual Studio 2022 and Unity 2021 LTS.
 
-New-Item -Path $windowsSdkDirectory -Force | Out-Null
+_Install Visual Studio Build Tools_
 
-New-ItemProperty -Path $windowsSdkDirectory `
-     -Name InstallationFolder `
-     -PropertyType String `
-     -Force `
-     -Value "C:\Program Files (x86)\Windows Kits\10\" | Out-Null
+The recommended way to install the correct dependencies on the host is with [Chocolatey](https://chocolatey.org/). This works with Windows 10/11, Windows Server, and Windows Server Core.
 
-New-ItemProperty -Path $windowsSdkDirectory `
-     -Name ProductVersion `
-     -PropertyType String `
-     -Force `
-     -Value $env:WINDOWS_SDK_VERSION | Out-Null
+```powershell
+choco install visualstudio2022-workload-vctools --no-progress -y
 ```
 
-_Mount list of volumes_
+There is a way to install the same dependencies from the graphical [Visual Studio Installer](https://visualstudio.microsoft.com/downloads/), but the workload component options in the installer frequently change.
+
+_The following directories need to be bind-mounted into the container_
 
 - C:\Program Files (x86)\Windows Kits
 - C:\Program Files (x86)\Microsoft Visual Studio
-- C:\Program Files\Microsoft Visual Studio
```

Comment:
This was added here on purpose and is referenced in the code somewhere for when switching to x64 architecture.

Reply:
All I can say is that this directory does not exist on my Windows Server 2019 x64 host after installing the minimum required build tools and adding it causes docker to fail to run.

Reply:
If it's breaking then lets remove it indeed.

Here's the [reference](https://github.com/game-ci/unity-builder/blob/579daa93a634c0d09499e5f907f2b42b4ff70fdb/src/model/platform-validation/validate-windows.ts#L45) I was referring to.

Reply:
Want to add a bit more context to this as I put that comment in. The default visual studio 2022 installation location is under the Program Files path which can be seen [here](https://learn.microsoft.com/en-us/visualstudio/install/change-installation-locations?view=vs-2022). For 2019 and below, the default install location is in Program Files x86. Are you installing a version of Visual Studio below 2022 or do you have build tools without installing the IDE? Is it something where a tool like chocolately installs to a different location? It does make sense that docker would fail if both paths are provided since you're likely to only have one visual studio installed so perhaps a note needs to be added about which path to mount.

Reply:
I'll add a note about visual studio 2019 and earlier. I think the missing info for me as what version numbers were affected. Thanks @AndrewKahr!